src/bin/main.rs

```diff
@@ -144,7 +147,11 @@ fn main() -> Result<()> {
                 OutputFormat::Json => if args.output_format.compact_output {
                     serde_json::ser::to_writer::<_, Value>(stdout(), &value)
                 } else {
-                    serde_json::ser::to_writer_pretty::<_, Value>(stdout(), &value)
+                    if args.output_format.raw_string_output && value.is_string() {
+                        stdout().write_all(value.unwrap_string().as_bytes()).or(Ok(()))
```

Comment:
I believe you added `.or(Ok(()))` because it doesn't compile without that. That's because the error type is different from the other `else` branch.
We're using `anyhow` crate to mitigate this various-different-error-type situation. In this case, we can wrap the error values to make them same by something like
```rust
if args.output_format.raw_string_output && value.is_string() {
    stdout()
        .write_all(value.unwrap_string().as_bytes())
        .with_context(|| "Failed to output")
} else {
    serde_json::ser::to_writer_pretty::<_, Value>(stdout(), &value)
        .with_context(|| "Write to output")
}
```

So can we
- kill L:156 (which used to tolerate error type difference between `serde_json` and `serde_yaml`),
- and copy that `.with_context` thing to 3 branches inside `OutputFormat::Json`?

Since 2 of those 3 are actually applied to the same type returned by `serde_json`, it's also possible to do
```
if ... {
  ....with_context(...)
} else {
  if {
  } else {
  }.with_context(...)
}
```
but it's up to you!

Reply:
Thanks for pointing this. Is there a way to bind `value` as type `Value` in this first if?
```
if args.output_format.raw_output && value.is_string() {
```

Reply:
Do you mean destructure-bind as `Value::String(s)`? I think we can't do it with `if` yet (https://github.com/rust-lang/rust/issues/53667), since we have the extra condition `args.output_format.raw_output`. (If there were not, we could've done `if let Value::String(s) = value { ... }`)

Though it's possible with `match`, like
```rust
match value {
    Value::String(s) if args.output_format.compact_output => {
    }
    _ => {
        // do something else
    }
}
```

Reply:
Thanks. Updated.